File: classifier/data/raw/human/samples/project_euler_zig/problem_094.zig
const std = @import("std");

// Project Euler 94: sum perimeters of almost-equilateral integral-area triangles.
pub fn main() !void {
    const limit: i64 = 1_000_000_000;
    var previous: i64 = 1; // Degenerate predecessor of (5, 5, 6).
    var equal_side: i64 = 5;
    var base_delta: i64 = 1; // The base is equal_side + base_delta.
    var sum: i64 = 0;
    while (true) {
        const perimeter = 3 * equal_side + base_delta;
        if (perimeter > limit) break;
        sum += perimeter;
        const next = 4 * equal_side - previous - 2 * base_delta;
        previous = equal_side;
        equal_side = next;
        base_delta = -base_delta;
    }
    var buffer: [32]u8 = undefined;
    try std.Io.File.writeStreamingAll(std.Io.File.stdout(), std.Io.Threaded.global_single_threaded.io(), (try std.fmt.bufPrint(&buffer, "{d}\n", .{sum})));
}

File: classifier/data/raw/ai/samples/project_euler_java/Problem094.java
// Project Euler 94: sum perimeters of almost-equilateral integral-area triangles.
public class Problem094 {
    public static void main(String[] args) {
        long m = 2, n = 0, total = 0;
        while (true) {
            long nextM = 2 * m + 3 * n, nextN = m + 2 * n; // multiply by 2 + sqrt(3)
            m = nextM; n = nextN; // m*m - 3*n*n = 4
            long side, base;
            if ((m + 1) % 3 == 0) { side = (m + 1) / 3; base = side + 1; }
            else { side = (m - 1) / 3; base = side - 1; }
            long perimeter = 2 * side + base;
            if (perimeter >= 1_000_000_000L) break;
            if (side > 1) total += perimeter;
        }
        System.out.println(total);
    }
}
